sympy/sets/tests/test_conditionset.py

```diff
@@ -85,38 +112,40 @@ def test_free_symbols():
         ).free_symbols == {z}
     assert ConditionSet(x, Eq(x, 0), FiniteSet(x, z)
         ).free_symbols == {x, z}
-    assert ConditionSet(x, Eq(x, 0), ImageSet(Lambda(y, y**2), S.Integers)
-        ).free_symbols == set()
+    assert ConditionSet(x, Eq(x, 0), ImageSet(Lambda(y, y**2),
+        S.Integers)).free_symbols == set()
 
 
 def test_bound_symbols():
     assert ConditionSet(x, Eq(y, 0), FiniteSet(z)
-        ).bound_symbols == {x}
+        ).bound_symbols == [x]
     assert ConditionSet(x, Eq(x, 0), FiniteSet(x, y)
-        ).bound_symbols == {x}
+        ).bound_symbols == [x]
     assert ConditionSet(x, x < 10, ImageSet(Lambda(y, y**2), S.Integers)
-        ).bound_symbols == {x}
+        ).bound_symbols == [x]
     assert ConditionSet(x, x < 10, ConditionSet(y, y > 1, S.Integers)
-        ).bound_symbols == {x}
+        ).bound_symbols == [x]
 
 
 def test_as_dummy():
-    _0 = Symbol('_0')
+    _0, _1 = symbols('_0 _1')
     assert ConditionSet(x, x < 1, Interval(y, oo)
         ).as_dummy() == ConditionSet(_0, _0 < 1, Interval(y, oo))
     assert ConditionSet(x, x < 1, Interval(x, oo)
         ).as_dummy() == ConditionSet(_0, _0 < 1, Interval(x, oo))
     assert ConditionSet(x, x < 1, ImageSet(Lambda(y, y**2), S.Integers)
         ).as_dummy() == ConditionSet(
             _0, _0 < 1, ImageSet(Lambda(_0, _0**2), S.Integers))
+    e = ConditionSet((x, y), x <= y, S.Reals**2)
+    assert e.bound_symbols == [x, y]
+    assert e.as_dummy() == ConditionSet((_0, _1), _0 <= _1, S.Reals**2)
+    assert e.as_dummy() == ConditionSet((y, x), y <= x, S.Reals**2
+        ).as_dummy()
 
 
 def test_subs_CondSet():
     s = FiniteSet(z, y)
     c = ConditionSet(x, x < 2, s)
-    # you can only replace sym with a symbol that is not in
-    # the free symbols
-    assert c.subs(x, 1) == c
     assert c.subs(x, y) == ConditionSet(y, y < 2, s)
```

Comment:
I don't think it should be possible to substitute the bound variable with subs like this. It should be necessary to use xreplace to change bound symbols.

Reply:
So only via the mask/xreplace/unmask idiom?

Reply:
Yes, I think so. Same as for Integral.